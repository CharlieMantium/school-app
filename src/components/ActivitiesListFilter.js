import React from 'react';
import Input from './Input';

const ActivitiesListFilter = () => (
  <div>
    <Input type="text" />
  </div>
);

export default ActivitiesListFilter;
