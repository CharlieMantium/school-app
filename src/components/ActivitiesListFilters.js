import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {setTextFilter} from '../actions/filters';

const ActivitiesListFilters = ({ filters: {text}, dispatch }) => (
  <div>
    <input type="text" value={text} onChange={(e) => {
      dispatch(setTextFilter(e.target.value));
    }}/>
  </div>
);

ActivitiesListFilters.propTypes = {
  filters: PropTypes.object,
  text: PropTypes.string,
  dispatch: PropTypes.func
};

const mapStateToProps = (state) => ({
  filters: state.filters
});

export default connect(mapStateToProps)(ActivitiesListFilters);
