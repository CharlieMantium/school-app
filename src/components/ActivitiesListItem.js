import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { EDIT_ACTIVITY_ROUTE } from '../constants/routes';

const ActivitiesListItem = ({ name, day, classNo, teacher, id }) => (
  <div>
    <Link to={EDIT_ACTIVITY_ROUTE(id)}>
      <h3>{name}</h3>
    </Link>
    <p>{`On ${day} on ${classNo} class with ${teacher}`}</p>
  </div>
);

ActivitiesListItem.propTypes = {
  name: PropTypes.string,
  day: PropTypes.string,
  classNo: PropTypes.string,
  teacher: PropTypes.string,
  id: PropTypes.string
};

export default ActivitiesListItem;
