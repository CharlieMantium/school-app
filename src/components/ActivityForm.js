import React from 'react';

export default class ActivityForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      classNo: props.activity ? props.activity.classNo : '',
      day: props.activity ? props.activity.day : '',
      name: props.activity ? props.activity.name : '',
      room: props.activity ? props.activity.room : '',
      teacher: props.activity ? props.activity.teacher : '',
      error: ''
    };
  };
  onNameChange = (e) => {
    const name = e.target.value;
    this.setState(() => ({name}));
  };
  onTeacherChange = (e) => {
    const teacher = e.target.value;
    this.setState(() => ({teacher}));
  };
  onRoomChange = (e) => {
    const room = e.target.value;
    this.setState(() => ({room}));
  };
  onClassNoChange = (e) => {
    const classNo = e.target.value;
    if (!classNo || classNo.match(/^[1-9]\d*$/)) {
      this.setState(() => ({classNo}));
    };
  };
  onDayChange = (e) => {
    const day = e.target.value;
    this.setState(() => ({day}));
  };
  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.name || !this.state.day || !this.state.classNo) {
      this.setState(() => ({error: 'Please provide Activity Name, Day and Class Number!'}))
    } else {
      this.setState(() => ({error: ''}))
      this.props.onSubmit({
        classNo: this.state.classNo,
        day: this.state.day,
        name: this.state.name,
        room: this.state.room,
        teacher: this.state.teacher,
      })
    }
  };
  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="Activity Name"
            autoFocus
            value={this.state.name}
            onChange={this.onNameChange}
          />
          <input
            type="text"
            placeholder="Teacher"
            value={this.state.teacher}
            onChange={this.onTeacherChange}
          />
          <input
            type="text"
            placeholder="Room"
            value={this.state.room}
            onChange={this.onRoomChange}
          />
          <input
            type="number"
            placeholder="Class Number"
            value={this.state.classNo}
            onChange={this.onClassNoChange}
          />
          <select
            value={this.state.day}
            onChange={this.onDayChange}
            required={true}
          >
            <option value=''>Pick a day</option>
            <option value={1}>Monday</option>
            <option value={2}>Tuesday</option>
            <option value={3}>Wednesday</option>
            <option value={4}>Thursday</option>
            <option value={5}>Friday</option>
            <option value={6}>Saturday</option>
            <option value={7}>Sunday</option>
          </select>
          <button>Add Activity</button>
        </form>
      </div>
    )
  }
}
