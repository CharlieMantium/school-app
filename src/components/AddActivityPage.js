import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import AcivityForm from './ActivityForm';
import { addActivity } from '../store/activities/actions';
import { ACTIVITY_PLAN_ROUTE } from '../constants/routes';
import historyPushPropTypeShape from '../prop-types/history';

export const AddActivityPage = ({ onAddActivity, history }) => (
  <div>
    <h1>Add Activity</h1>
    <AcivityForm
      onSubmit={activity => {
        onAddActivity(activity);
        history.push(ACTIVITY_PLAN_ROUTE);
      }}
    />
  </div>
);

AddActivityPage.propTypes = {
  onAddActivity: PropTypes.func.isRequired,
  history: PropTypes.shape(historyPushPropTypeShape).isRequired,
};

const mapDispatchToProps = {
  onAddActivity: addActivity,
};

export default connect(
  null,
  mapDispatchToProps,
)(AddActivityPage);

// TODO: Ask Lukasz about null in export

// TODO: handle absolute imports
