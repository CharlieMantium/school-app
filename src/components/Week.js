import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import activityPropTypeShape from '../prop-types/activity';
import Day from './Day';
import { daysOfTheWeek } from '../constants/dates';
import { getActivityItems, getActivitiesForDay } from '../store/activities/selectors';

export const Week = ({ activities }) => (
  <div>
    {daysOfTheWeek.map(day => (
      <Day weekDay={day} activities={getActivitiesForDay(activities, day)} key={day} />
    ))}
  </div>
);

Week.propTypes = {
  activities: PropTypes.arrayOf(PropTypes.shape(activityPropTypeShape)),
};

Week.defaultProps = {
  activities: [],
};

const mapStateToProps = state => ({
  activities: getActivityItems(state),
});

export default connect(mapStateToProps)(Week);
