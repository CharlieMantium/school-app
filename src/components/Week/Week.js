import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';

import activityPropTypeShape from 'prop-types/activity';
import { daysOfTheWeek } from 'constants/dates';
import { getActivityItems, getActivitiesForDay } from 'store/activities/selectors';
import { spacing } from 'styles/base/base';

import Day from '../Day';

const StyledWeekWrapper = styled.div`
  @media (min-width: ${spacing.desktopBreakpoint}) {
    align-items: stretch;
    display: flex;
  }
`;

const Week = ({ activities }) => (
  <StyledWeekWrapper>
    {daysOfTheWeek.map(day => (
      <Day
        weekDay={day}
        activities={getActivitiesForDay(activities, day)}
        key={day}
        data-test="day-component"
      />
    ))}
  </StyledWeekWrapper>
);

Week.propTypes = {
  activities: PropTypes.arrayOf(PropTypes.shape(activityPropTypeShape)),
};

Week.defaultProps = {
  activities: [],
};

const mapStateToProps = state => ({
  activities: getActivityItems(state),
});

export { Week as WeekUnwrapped };
export default connect(mapStateToProps)(Week);
