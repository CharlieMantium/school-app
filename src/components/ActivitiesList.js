import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import ActivitiesListItem from './ActivitiesListItem';
import selectActivities from '../store/activities/selectors/activities';

const ActivitiesList = ({ activities }) => (
  <div>
    <h1>Activities List</h1>
    {activities.map((activity) => (
      <ActivitiesListItem {...activity} key={activity.id}/>
    ))}
  </div>
);

ActivitiesList.propTypes = {
  activities: PropTypes.array
};

const mapStateToProps = (state) => ({
  activities: selectActivities(state.activities, state.filters)
});

export default connect(mapStateToProps)(ActivitiesList);
