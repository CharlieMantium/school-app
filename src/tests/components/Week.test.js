import React from 'react';
import { shallow } from 'enzyme';

import { Week } from '../../components/Week';
import testState from '../fixtures/state';

test('should render Week correctly', () => {
  const wrapper = shallow(<Week activities={testState.activities.items} />);
  expect(wrapper).toMatchSnapshot();
});
