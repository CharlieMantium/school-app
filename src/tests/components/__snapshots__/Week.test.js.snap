// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Week should render Week correctly 1`] = `
<div>
  <Day
    activities={
      Array [
        Object {
          "classNo": "1",
          "day": "monday",
          "id": "2",
          "name": "polski",
          "room": "hell",
          "teacher": "Malski",
        },
        Object {
          "classNo": "2",
          "day": "monday",
          "id": "3",
          "name": "polski",
          "room": "103",
          "teacher": "Malski",
        },
      ]
    }
    key="monday"
    weekDay="monday"
  />
  <Day
    activities={
      Array [
        Object {
          "classNo": "3",
          "day": "tuesday",
          "id": "1",
          "name": "matma",
          "room": "gym",
          "teacher": "Indiana Jones",
        },
      ]
    }
    key="tuesday"
    weekDay="tuesday"
  />
  <Day
    activities={
      Array [
        Object {
          "classNo": "7",
          "day": "wednesday",
          "id": "4",
          "name": "w-f",
          "room": "gym",
          "teacher": "RAMBO",
        },
      ]
    }
    key="wednesday"
    weekDay="wednesday"
  />
  <Day
    activities={Array []}
    key="thursday"
    weekDay="thursday"
  />
  <Day
    activities={
      Array [
        Object {
          "classNo": "5",
          "day": "friday",
          "id": "5",
          "name": "WOS",
          "room": "nowhere",
          "teacher": "Chopin",
        },
      ]
    }
    key="friday"
    weekDay="friday"
  />
  <Day
    activities={Array []}
    key="saturday"
    weekDay="saturday"
  />
  <Day
    activities={Array []}
    key="sunday"
    weekDay="sunday"
  />
</div>
`;
